app/data_models/supplementary_data_store.py

```diff
@@ -94,17 +94,31 @@ def get_data(
         For example if you wanted the identifier for the first item in "some_list"
         it would be get_data(identifier="some_list", selectors=["identifier"], list_item_id=list_item_id-1)
         """
+        if self.is_data_repeating(identifier) and not list_item_id:
+            raise InvalidSupplementaryDataSelector(
+                f"Cannot reference items from `{identifier}` outside a repeat"
+            )
+
         value = self._data_map.get((identifier, list_item_id))
-        # for nested data, index with each selector
+        # for nested data, index with each selector, or return None if there is no data to index
         for selector in selectors or []:
```

Comment:
Just having a quick glance while reviewing mock sds. I know it's not this PR but can you explain the logic here? if `selectors` is false then you iterate through empty list so don't iterate at all?

Reply:
It was a suggestion made on a previous PR to reduce the amount of nesting, so that I wouldn't need
```
if selectors:
 for selector in selectors:
   # extra indent
```
and could have the one liner instead which I think is quite tidy, but I don't mind too much what we go with, can update if preferred